packages/lesswrong/components/posts/PostsPage/PostsPodcastPlayer.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import NoSSR from '@material-ui/core/NoSsr';
+import { registerComponent } from '../../../lib/vulcan-lib';
+import { applePodcastIcon } from '../../icons/ApplePodcastIcon';
+import { spotifyPodcastIcon } from '../../icons/SpotifyPodcastIcon';
+import { isClient } from '../../../lib/executionEnvironment';
+
+const styles = (): JssStyles => ({
+  embeddedPlayer: {
+    marginBottom: '2px'
+  },
+  podcastIconList: {
+    paddingLeft: '0px',
+    marginTop: '0px'
+  },
+  podcastIcon: {
+    display: 'inline-block',
+    marginRight: '8px'
+  }
+});
+
+const PostsPodcastPlayer = ({ podcastEpisode, classes }: {
+  podcastEpisode: PostsBase_podcastEpisode,
+  classes: ClassesType
+}) => {
+  const embedScriptFunction = (src: string, clientDocument: Document) => <>{
+    function(d) {
+      const x = d.getElementById('buzzsproutPlayerScript');
+      if (x) x.parentNode?.removeChild(x);
+      const s=d.createElement('script');
+      d.head.appendChild((
+        s.async=true,
+        s.src=src,
+        s.id='buzzsproutPlayerScript',
+        s
+      ));
+    }(clientDocument)
+  }</>;
+
+  return <>
+    <div
+      id={`buzzsprout-player-${podcastEpisode.externalEpisodeId}`}
+      className={classes.embeddedPlayer}
+    />
+    {isClient && <NoSSR>
```

Comment:
For some reason I was getting client/server render-mismatch errors without it.  Let me reproduce.

Reply:
Yeah, without the `isClient`, I get the following errors in the console:
```
Warning: Expected server HTML to contain a matching <div> in <div>.
  ...
Uncaught Error: Hydration failed because the initial UI does not match what was rendered on the server.
  ...
```

as well as an error on the server: `ReferenceError: document is not defined`.  This makes it seem like `NoSSR` isn't actually doing the thing it's supposed to be doing?  Definitely a bit confused as to what's going on there.